File: appJava/src/main/java/com/brandon/appjava/viewmodel/createaccount/LegacyCreateAccountViewModel.java
package com.brandon.appjava.viewmodel.createaccount;

import static com.brandon.appjava.utilities.ConnectionErrors.NO_CONNECTION_ERROR;
import static com.brandon.appjava.utilities.UserInputErrors.NO_INPUT_ERROR;
import static com.brandon.appjava.viewmodel.createaccount.CreateAccountUiStates.IDLE;

import android.net.ConnectivityManager;

import com.brandon.appjava.domain.user.CreateUserUseCaseJava;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;
import io.reactivex.rxjava3.annotations.NonNull;
import io.reactivex.rxjava3.core.BackpressureStrategy;
import io.reactivex.rxjava3.core.Flowable;
import io.reactivex.rxjava3.subjects.BehaviorSubject;

@HiltViewModel
public class LegacyCreateAccountViewModel extends CreateAccountViewModel {
    private final CreateUserUseCaseJava createUserUseCaseJava;
    private final ConnectivityManager connectivityManager;

    private final BehaviorSubject<CreateAccountUiStateJava> _uiState = BehaviorSubject.createDefault(new CreateAccountUiStateJava(IDLE, NO_CONNECTION_ERROR, "", "", "", "", "", NO_INPUT_ERROR, NO_INPUT_ERROR, NO_INPUT_ERROR, NO_INPUT_ERROR, NO_INPUT_ERROR));

    public @NonNull Flowable<CreateAccountUiStateJava> getState() {
        return _uiState.toFlowable(BackpressureStrategy.LATEST);
    }

    @Inject
    public LegacyCreateAccountViewModel(CreateUserUseCaseJava createUserUseCaseJava, ConnectivityManager connectivityManager) {
        this.createUserUseCaseJava = createUserUseCaseJava;
        this.connectivityManager = connectivityManager;
    }

    public void firstNameChanged(String name) {
        _uiState.getValue().setFirstName(name);
    }

    public void lastNameChanged(String name) {
        _uiState.getValue().setLastName(name);
    }

    public void emailChanged(String email) {
        _uiState.getValue().setEmail(email);
    }

    public void dateChanged(String date) {
        _uiState.getValue().setDate(date);
    }

    public void locationChanged(String location) {
        _uiState.getValue().setLocation(location);
    }


    @Override
    protected CreateUserUseCaseJava createUserUseCaseJava() {
        return createUserUseCaseJava;
    }

    @Override
    protected ConnectivityManager connectivityManager() {
        return connectivityManager;
    }

    @Override
    protected void setState(CreateAccountUiStateJava newState) {
        _uiState.onNext(newState);
    }
}

File: createAccountLegacy/src/main/java/com/brandon/createaccountlegacy/viewmodel/LegacyCreateAccountViewModel.kt
package com.brandon.createaccountlegacy.viewmodel

import android.util.Log
import androidx.lifecycle.viewModelScope
import com.brandon.createaccount.core.usecase.CreateUserUseCase
import com.brandon.createaccount.core.viewmodel.CreateAccountViewModel
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch
import javax.inject.Inject

@HiltViewModel
class LegacyCreateAccountViewModel @Inject constructor(
    override val createUserUseCase: CreateUserUseCase
): CreateAccountViewModel() {

    override var state: CreateAccountUiState
        get() = _uiState.value
        set(value) { _uiState.value = value }

    private var _uiState: MutableStateFlow<CreateAccountUiState> = MutableStateFlow(CreateAccountUiState.Idle())
    val uiState: StateFlow<CreateAccountUiState>
        get() = _uiState.asStateFlow()

}